WPFHospitalEditor/AdditionalInformation.xaml.cs

```diff
@@ -93,52 +90,71 @@ private void ModifyDynamicWPFGrid(string[] contentRows)
+            } else
+            {
+                for (int i = 0; i < contentRows.Length; i++)
+                {
+                    if (contentRows[i].Equals("")) break;
+                    setRowContent(i);
+                }
             }
             insertData();
             addCancelButton(contentRows);
             addOkButton(contentRows);
         }
 
+        private bool PatientIsLogged()
+        {
+            if (role.Equals("patient")) return true;
+            return false;
+        }
+
+        private bool StringContainsWorkingHours(string str)
+        {
+            if (str.Contains("Working Hours")) return true;
+            return false;
+        }
+
+        private void setRowContent(int row)
+        {
+            string[] label = contentRows[row].Split("=");
+            labelContent.Add(label[0]);
+            value.Add(label[1]);
+        }
+
         private void addOkButton(string[] contentRows)
         {
             Button ok = new Button();
-            ok.Content = "OK";
-            ok.BorderThickness = new Thickness(0);
             ok.HorizontalAlignment = HorizontalAlignment.Right;
-            ok.VerticalAlignment = VerticalAlignment.Center;
-            ok.Background = Brushes.SkyBlue;
-            ok.Width = buttonWidth;
-            ok.Height = buttonHeight;
-            ok.Foreground = Brushes.White;
             ok.Click += Done_Click;
             Grid.SetRow(ok, contentRows.Length + 2);
             Grid.SetColumn(ok, 2);
-            DynamicGrid.Children.Add(ok);
+            setButtonsCommonAttributes(ok, "Ok");
         }
 
         private void addCancelButton(string[] contentRows)
         {
             Button cancel = new Button();
-            cancel.Content = "Cancel";
-            cancel.BorderThickness = new Thickness(0);
             cancel.HorizontalAlignment = HorizontalAlignment.Left;
-            cancel.VerticalAlignment = VerticalAlignment.Center;
-            cancel.Background = Brushes.SkyBlue;
-            cancel.Width = buttonWidth;
-            cancel.Height = buttonHeight;
-            cancel.Foreground = Brushes.White;
             cancel.Click += Close_Click;
             Grid.SetRow(cancel, contentRows.Length + 2);
             Grid.SetColumn(cancel, 1);
-            DynamicGrid.Children.Add(cancel);
+            setButtonsCommonAttributes(cancel, "Cancel");
+        }
+
+        private void setButtonsCommonAttributes(Button button, string name)
+        {
+            button.BorderThickness = new Thickness(0);
+            button.Content = name;
+            button.VerticalAlignment = VerticalAlignment.Center;
+            button.Background = Brushes.SkyBlue;
+            button.Width = AllConstants.additionalInformationsbuttonWidth;
+            button.Height = AllConstants.additionalInformationsbuttonHeight;
+            button.Foreground = Brushes.White;
+            DynamicGrid.Children.Add(button);
```

Comment:
If you call this function in only once, you could define all this properties in xaml file (not xaml.cs)